### PR TITLE
feat(watchAtMost): new function

### PR DESCRIPTION
**Pull request by webfansplz**
see #762

### CODE REVIEW

**Comment by antfu** [comment]
Looks interesting, but can you share a bit context of why this is useful?

**Comment by webfansplz** [comment]
My initial idea was watchOnce (I use it a lot in my projects，but with watch i need to manually call the stop function). i use watchWithCount instead of watchOnce that because i think it's more flexible.

**Comment by patak-js** [comment]
Maybe `watchN(4, ...)` or `watchAtMost(4, ...)` to be closer to `watchOnce`. A `watchUntil(ms,...)` may also be interesting 🤔

**Comment by patak-js** [comment]
Do you have a concrete example where you used this @webfansplz? I'm also curious about your real use cases

**Comment by webfansplz** [comment]
```ts

const { stop } = watch(
                props.response!,
                () => {
                    nextTick(() => {
                        const floor = route.query.floor
                        onToggleActiveFloor(Number(floor))
                        stop()
                    })
                },
                { immediate: true },
            )

```
This is one of the scenes: I need to listen for data responses and do UI initialization with route parameters.

**Comment by webfansplz** [comment]
> Maybe `watchN(4, ...)` or `watchAtMost(4, ...)` to be closer to `watchOnce`. A `watchUntil(ms,...)` may also be interesting 🤔

i agree with you. maybe `watchAtMost(4, ...)` is more appropriate ~

File: packages/shared/watchAtMost/index.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  options: WatchAtMostOptions<Immediate> = { max: 1 },
+): WatchAtMostReturn {
+  const {
+    max,
+    ...watchOptions
+  } = options
+  const { count, filter } = countFilter(max)
+  const stop = watch(
+    source,
+    createFilterWrapper(filter, (...args) => {
```
Comment (antfu) [inline packages/shared/watchAtMost/index.ts:32]: I don't think the filter is needed here (a bit overengineered from my point of view. Simply count and stop will be good enough.

**Review by antfu — CHANGES_REQUESTED** [review]
I think it would be great to also include a shorthand `watchOnce` that uses this function under the hood

**Comment by antfu** [comment]
I will made the changes directly

**Comment by webfansplz** [comment]
@antfu thanks~

**Pull request merged**